Sources/XcodeGraphMapper/Mappers/Targets/PBXTargetMapper.swift

```diff
@@ -331,6 +353,166 @@ struct PBXTargetMapper: PBXTargetMapping {
+
+            let attributesByRelativePath = fileSystemSynchronizedGroup.exceptions?.reduce([:]) { acc, element in
+                acc.merging(element.attributesByRelativePath ?? [:], uniquingKeysWith: { $1 })
+            }
+
+            let groupFrameworks: [TargetDependency] = try await fileSystem.glob(
+                directory: directory,
+                include: [
+                    "**/*.framework",
+                ]
+            )
+            .collect()
+            .filter {
+                !membershipExceptions.contains($0.relative(to: directory).pathString)
+            }
+            .map {
+                return .framework(
+                    path: $0,
+                    status: attributesByRelativePath?[$0.relative(to: directory).pathString]?
+                        .contains("Weak") == true ? .optional : .required,
+                    condition: nil
+                )
+            }
+            frameworks.append(contentsOf: groupFrameworks)
+        }
+
+        return frameworks
+    }
+
+    private func addHeadersFromFileSystemSynchronizedGroups(
+        from pbxTarget: PBXTarget,
+        xcodeProj: XcodeProj,
+        headers: Headers?
+    ) async throws -> Headers? {
+        let fileSystemSynchronizedGroups = pbxTarget.fileSystemSynchronizedGroups ?? []
+        var publicHeaders = headers?.public ?? []
+        var privateHeaders = headers?.private ?? []
+        var projectHeaders = headers?.project ?? []
+        for fileSystemSynchronizedGroup in fileSystemSynchronizedGroups {
+            guard let path = fileSystemSynchronizedGroup.path else { continue }
+            let directory = xcodeProj.srcPath.appending(component: path)
+            for synchronizedBuildFileSystemExceptionSet in fileSystemSynchronizedGroup.exceptions ?? [] {
+                for publicHeader in synchronizedBuildFileSystemExceptionSet.publicHeaders ?? [] {
+                    publicHeaders.append(directory.appending(component: publicHeader))
+                }
+                for privateHeader in synchronizedBuildFileSystemExceptionSet.privateHeaders ?? [] {
+                    privateHeaders.append(directory.appending(component: privateHeader))
+                }
+            }
+
+            let groupHeaders = try await fileSystem.glob(
+                directory: directory,
+                include: [
+                    "**/*.{h,hpp}",
+                ]
+            )
+            .collect()
+            .filter {
+                !privateHeaders.contains($0) && !publicHeaders.contains($0)
+            }
+            projectHeaders.append(contentsOf: groupHeaders)
+        }
+        if !publicHeaders.isEmpty || !privateHeaders.isEmpty || !projectHeaders.isEmpty || headers != nil {
+            return Headers(
+                public: publicHeaders,
+                private: privateHeaders,
+                project: projectHeaders
+            )
+        } else {
+            return headers
+        }
+    }
 }
 
```

Comment:
Since multiple functions use a similar globbing pattern with membership exception filtering, we could extract this logic into a helper function.

```swift
/// Performs a glob search in the given directory using specified patterns, filtering out paths
/// that appear in the membershipExceptions set.
private func globFiles(
    in directory: Path,
    patterns: [String],
    membershipExceptions: Set<String> = []
) async throws -> [Path] {
    // The glob patterns are built as comma-separated lists of file extensions.
    return try await fileSystem.glob(
        directory: directory,
        include: patterns
    )
    .collect()
    .filter { !membershipExceptions.contains($0.relative(to: directory).pathString) }
}
```